server/datastore/mysql/vpp.go

```diff
@@ -253,9 +297,24 @@ func (ds *Datastore) SetTeamVPPApps(ctx context.Context, teamID *uint, appFleets
 
 	for _, appFleet := range appFleets {
 		// upsert it if it does not exist or SelfService or InstallDuringSetup flags are changed
-		if existingFleet, ok := existingApps[appFleet.VPPAppID]; !ok || existingFleet.SelfService != appFleet.SelfService ||
+		existingFleet, ok := existingApps[appFleet.VPPAppID]
+		var labelsChanged bool
+		if ok {
+			existingLabels, err := ds.getExistingLabels(ctx, appFleet.AppTeamID)
+			if err != nil {
+				return ctxerr.Wrap(ctx, err, "getting existing labels for vpp app")
+			}
+
+			labelsChanged = !existingLabels.Equal(appFleet.ValidatedLabels)
+		}
+
+		if !ok ||
+			existingFleet.SelfService != appFleet.SelfService ||
+			labelsChanged ||
 			appFleet.InstallDuringSetup != nil &&
```

Comment:
On first glance we want a stronger check here, but I guess we still support defining install-on-setup in two places in YAML until label enforcement on setup experience gets implemented (I think I saw a YAML change requested there).